pymc3/model.py

```diff
@@ -1518,3 +1493,324 @@ def all_continuous(vars):
         return False
     else:
         return True
+
+
+class ConstantNodeException(Exception):
```

Comment:
Maybe move these to a separate file?

Reply:
`not_shared_or_constant_variable` needs `model.FreeRV`, `model.MultiObservedRV` and `model.TransformedRV`. If we move it away, we'll get into a circular import problem. That's why I left it in `model.py`.